### ISSUE TITLE
Add "input" label mode to tf.keras.preprocessing.image_dataset_from_directory

### ISSUE THREAD

**Issue by FTC55** [keras-team/keras#15750]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (you are using): 2.4.0
Are you willing to contribute it (Yes/No) : No

**Describe the feature and the current behavior/state**.
Currently, there is no way to get `tf.keras.preprocessing.image_dataset_from_directory` to create a dataset the labels of which are the images themselves, for use in autoencoders. The behavior would be the same as that of `tf.keras.preprocessing.image.ImageDataGenerator.flow_from_directory` with the argument `class_mode="input"`. Perhaps the preexisting argument `label_mode` could be modified to include an `"input"` option.

**Will this change the current api? How?**
Minimal change to `tf.keras.preprocessing.image_dataset_from_directory` is required. Because of the nature of what is essentially a very small addition, it would be extremely unlikely to pose a compatibility issue.

**Who will benefit from this feature?**
Users wishing to benefit from the advantages of the `tf.data` API to train their image data autoencoders or more broadly, for any of the use cases already satisfied by this feature in `tf.keras.preprocessing.image.ImageDataGenerator.flow_from_directory`.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Comment by haifeng-jin** [comment]
I assume this `tf.keras.preprocessing.image_dataset_from_directory` is a shortcut only for the most common use cases.
For any other use cases, one should use `Sequence` or Python generator to load from the disk.
